arcgis-ios-sdk-samples/Features/Filter by definition expression or display filter/DefinitionExpressionDisplayFilterViewController.swift

```diff
@@ -0,0 +1,108 @@
+    /// The feature layer made with the feature service URL.
+    let featureLayer = AGSFeatureLayer(featureTable: AGSServiceFeatureTable(url: featureServiceURL!))
+    /// The display filter definition to apply to the feature layer.
+    var displayFilterDefinition: AGSDisplayFilterDefinition?
+    /// The definition expression to apply to the feature layer.
+    var definitionExpression: String?
+    
+    /// Applies the definition expression.
+    @IBAction func applyDefinitionExpression() {
+        // Set the definition expression.
+        displayFilterDefinition = nil
+        definitionExpression = "req_Type = 'Tree Maintenance or Damage'"
+        countFeatures()
+    }
+    
+    /// Applies the display filter
+    @IBAction func applyFilter() {
+        definitionExpression = ""
+        // Create a display filter with a name and an SQL expression.
+        guard let damagedTrees = AGSDisplayFilter(name: "Damaged Trees", whereClause: "req_type LIKE '%Tree Maintenance%'") else { return }
+        // Set the manual display filter definition using the display filter.
+        let manualDisplayFilterDefinition = AGSManualDisplayFilterDefinition(activeFilter: damagedTrees, availableFilters: [damagedTrees])
+        // Apply the display filter definition.
+        displayFilterDefinition = manualDisplayFilterDefinition
+        countFeatures()
+    }
+    
+    /// Reset the definition expression.
+    @IBAction func resetDefinitionExpression() {
+        definitionExpression = ""
+        displayFilterDefinition = nil
+        countFeatures()
+    }
+    
+    /// Create a map and set its attributes.
+    func makeMap() -> AGSMap {
+        // Initialize the map with the topographic basemap style.
+        let map = AGSMap(basemapStyle: .arcGISTopographic)
+        // Set the initial viewpoint.
+        let viewpoint = AGSViewpoint(latitude: 37.772296660953138, longitude: -122.44014487516885, scale: 100000)
+        map.initialViewpoint = viewpoint
+        // Add the feature layer to the map's operational layers.
+        map.operationalLayers.add(featureLayer)
+        // Load the map.
+        map.load { [weak self] error in
+            if let error = error {
+                self?.presentAlert(error: error)
+            }
+        }
+        return map
+    }
+    
+    /// Count the features according to the applied expressions.
+    func countFeatures() {
+        // Set the extent to the current view.
+        let extent = mapView.currentViewpoint(with: .boundingGeometry)?.targetGeometry.extent
+        // Create the query parameters and set its geometry.
+        let queryParameters = AGSQueryParameters()
+        queryParameters.geometry = extent
+        // Apply the expressions to the feature layer.
+        featureLayer.displayFilterDefinition = displayFilterDefinition
+        featureLayer.definitionExpression = definitionExpression!
```

Comment:
If you make `definitionExpression` non-optional, you don't need the force unwrap.
```suggestion
        featureLayer.definitionExpression = definitionExpression
```